domain/application/state/application.go

```diff
@@ -369,6 +371,55 @@ func (st *ApplicationState) deleteSimpleApplicationReferences(ctx context.Contex
 	return nil
 }
 
+// GetSecretsForOwners returns the secrets owned by the specified apps and/or units.`
+func (st *ApplicationState) GetSecretsForOwners(
+	ctx domain.AtomicContext, appOwners domainsecret.ApplicationOwners, unitOwners domainsecret.UnitOwners,
+) ([]*secrets.URI, error) {
+	if len(appOwners) == 0 && len(unitOwners) == 0 {
+		return nil, errors.New("must supply at least one app owner or unit owner")
+	}
+
+	query := `
+SELECT sm.secret_id AS &secretID.id
```

Comment:
I'd create a view for this and use it in both places.

Reply:
I was looking at this today and realised that this is a _movement_ of the service from secret to application. Can you propose a patch with just that part?